### DISCUSSION TITLE
createContext 시 null 을 허용할 것인지

### DISCUSSION THREAD

**Discussion by healtheloper** [Co-Studo/Co-Studo-front#33] [Polls]
@jindonyy @hemudi @mina-gwak 

해당 Discussion 을 연 이유

- 벨로퍼트님 글 [TypeScript 와 Context API 활용하기](https://react.vlpt.us/using-typescript/04-ts-context.html) 의  Comment
- [context check](https://github.com/Co-Studo/Co-Studo-front/pull/24#discussion_r983058379)

### 장점
- 사용자가 Context Provider 로 감싸고 사용해야할 컴포넌트를 감싸지 않았을 경우 context 를 찾지 못하는 로직을 쉽게 구현할 수 있음

```tsx
const useTabContext = () => {
  const context = useContext(TabContext);
  if (!context) {
    throw new Error('useTabContext should be used within TabContext.Provider');
  }
  return context;
};

```

### 단점

- useContext hook 을 만들어 사용할 시 null 일 수있다는 타입 에러로 인해 문제가 생길 수 있음 ( 정확히 어떤 상황인지 기록해둔게 없네요.. )

**Comment by hemudi** [comment]
null을 허용하는데 찬성합니다! ^_^)b

'[How to use React Context effectively](https://kentcdodds.com/blog/how-to-use-react-context-effectively) - [번역글](https://goongoguma.github.io/2021/06/05/How-to-use-React-Context-effectively/)' 글을 보면 Context 를 사용할 때 기본값이 유용한 경우가 생각보다 적고, 주로 Provider 로 감싸여져 있는지, 아닌지를 체크하는 용도로 사용 된다고 하더라구요. 저희도 실질적으로는 그러한 용도로 사용하고 있기 때문에 [Radio Componenet PR 코멘트](https://github.com/Co-Studo/Co-Studo-front/pull/24#discussion_r980975685)로 의견을 나눴듯이 undefined 보다는 null을 이용하여 좀 더 명시적으로 체크하는 것이 좋다고 생각하여 찬성합니다!
